.gitlab/install_ddtrace_deps.rb

```diff
@@ -0,0 +1,78 @@
+# require "fileutils"
+require "open3"
+require "bundler"
+
+lock_file_path = "./Gemfile.lock"
+install_dir = "./vendor"
+
+# lock_file_path = "/opt/datadog/apm/library/ruby/Gemfile.lock"
+# install_dir = "./vendor"
+
+lock_file_parser = Bundler::LockfileParser.new(Bundler.read_file(lock_file_path))
+
+gem_version_mapping = lock_file_parser.specs.each_with_object({}) do |spec, hash|
+  hash[spec.name] = spec.version.to_s
+  hash
+end
+
+def install_ddtrace_deps_without_native_extensions(gem_version_mapping: ,location: )
```

Comment:
I think this method name is misleading, `install_ddtrace_deps_without_native_extensions`, because `libdatadog` and `libddwaf` **do have** native extensions, but they are already precompiled for the user:
![Screenshot 2023-07-04 at 10 33 04 AM](https://github.com/DataDog/dd-trace-rb/assets/583503/10950857-7537-4c2b-8096-e98f6eb25c76).

I think a better name would be something like: `install_ddtrace_deps_without_local_compilation`.

Same for `install_ddtrace_deps_with_native_extensions`: `install_ddtrace_deps_with_local_compilation`.

Reply:
Same kind of language should be applied for `INSTALL_DDTRACE_NON_NATIVE_DEPS` as well.